Bob fails to run on Windows from a path containing spaces

**Describe the bug (REQUIRED)**
Bob fails with exception java.nio.file.NoSuchFileException if run from a folder with spaces in the path.

**To Reproduce (REQUIRED)**
Run Bob from a folder containing spaces, e.g. (arguments removed for clarity)
```java -jar "D:\User Documents\Richard\Defold projects\defold-playable-ads\playable_ad\build\bob_1f57126.jar" ```

**Expected behavior (REQUIRED)**
Bob runs without error

**Defold version (REQUIRED):**
1.2.184 (1f57126)

**Platforms (REQUIRED):**
 - Platforms: Windows
 - OS: Windows 10

**Logs (OPTIONAL):**
```java -jar "D:\User Documents\Richard\Defold projects\defold-playable-ads\playable_ad\build\bob_1f57126.jar" --email foo@bar.com --auth 12345 --build-server https://build.defold.com --variant release --texture-compression true --bundle-output playable_ad/build/output_js-web --platform js-web --archive distclean resolve build bundle

D:\User Documents\Richard\Defold projects\defold-playable-ads>java -jar "D:\User Documents\Richard\Defold projects\defold-playable-ads\playable_ad\build\bob_1f57126.jar" --email foo@bar.com --auth 12345 --build-server https://build.defold.com --variant release --texture-compression true --bundle-output playable_ad/build/output_js-web --platform js-web --archive distclean resolve build bundle
Download archive(s)Downloading 0 archive(s)
Exception in thread "main" java.lang.RuntimeException: java.nio.file.NoSuchFileException: D:\User%20Documents\Richard\Defold%20projects\defold-playable-ads\playable_ad\build\bob_1f57126.jar
        at com.dynamo.bob.ClassLoaderResourceScanner.scan(ClassLoaderResourceScanner.java:100)
        at com.dynamo.bob.fs.ClassLoaderMountPoint.walk(ClassLoaderMountPoint.java:107)
        at com.dynamo.bob.fs.AbstractFileSystem.walk(AbstractFileSystem.java:108)
        at com.dynamo.bob.Project.findSources(Project.java:1533)
        at com.dynamo.bob.Bob.setupProject(Bob.java:485)
        at com.dynamo.bob.Bob.mainInternal(Bob.java:542)
        at com.dynamo.bob.Bob.main(Bob.java:681)
Caused by: java.nio.file.NoSuchFileException: D:\User%20Documents\Richard\Defold%20projects\defold-playable-ads\playable_ad\build\bob_1f57126.jar
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at java.base/sun.nio.fs.WindowsFileAttributeViews$Basic.readAttributes(WindowsFileAttributeViews.java:53)
        at java.base/sun.nio.fs.WindowsFileAttributeViews$Basic.readAttributes(WindowsFileAttributeViews.java:38)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.readAttributes(WindowsFileSystemProvider.java:194)
        at java.base/java.nio.file.Files.readAttributes(Files.java:1763)
        at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1222)
        at java.base/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:726)
        at java.base/java.util.zip.ZipFile$CleanableResource.get(ZipFile.java:843)
        at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:246)
        at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:176)
        at java.base/java.util.jar.JarFile.<init>(JarFile.java:346)
        at java.base/java.util.jar.JarFile.<init>(JarFile.java:317)
        at java.base/java.util.jar.JarFile.<init>(JarFile.java:256)
        at com.dynamo.bob.ClassLoaderResourceScanner.scanJar(ClassLoaderResourceScanner.java:68)
        at com.dynamo.bob.ClassLoaderResourceScanner.scan(ClassLoaderResourceScanner.java:96)
        ... 6 more
```

**Workaround (OPTIONAL):**
Install Bob in a path without spaces.

--- a/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/ClassLoaderResourceScanner.java
+++ b/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/ClassLoaderResourceScanner.java
@@ -18,6 +18,7 @@ import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URL;
+import java.net.URLDecoder;
 import java.util.Enumeration;
 import java.util.HashSet;
 import java.util.Set;
@@ -66,7 +67,7 @@ public class ClassLoaderResourceScanner implements IResourceScanner {
 
     private static void scanJar(URL resource, String filter, Set<String> results) throws IOException {
         String resPath = resource.getPath();
-        String jarPath = resPath.replaceFirst("[.]jar[!].*", ".jar").replaceFirst("file:", "");
+        String jarPath = URLDecoder.decode(resPath.replaceFirst("[.]jar[!].*", ".jar"), "UTF-8").replaceFirst("file:", "");
         try (JarFile jarFile = new JarFile(jarPath)){
             Enumeration<JarEntry> entries = jarFile.entries();
             while(entries.hasMoreElements()) {

--- a/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/ClassLoaderResourceScanner.java
+++ b/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/ClassLoaderResourceScanner.java
@@ -67,6 +67,8 @@ public class ClassLoaderResourceScanner implements IResourceScanner {
 
     private static void scanJar(URL resource, String filter, Set<String> results) throws IOException {
         String resPath = resource.getPath();
+
+        // resPath is URL-encoded, e.g. space characters represented as %20, so decode before deriving file paths from it
         String jarPath = URLDecoder.decode(resPath.replaceFirst("[.]jar[!].*", ".jar"), "UTF-8").replaceFirst("file:", "");
         try (JarFile jarFile = new JarFile(jarPath)){
             Enumeration<JarEntry> entries = jarFile.entries();

